public/model/layerRenderController.ts

```diff
@@ -22,16 +22,25 @@ interface MaplibreRef {
   current: Maplibre | null;
 }
 
-// OpenSearch only accepts longitude in range [-180, 180]
-// Maplibre could return value out of the range
-function adjustLongitudeForSearch(lon: number) {
-  if (lon < -180) {
-    return -180;
-  }
-  if (lon > 180) {
-    return 180;
+// calculate lng limits based on map bounds
+// maps can render more than 1 copies of map at lower zoom level and displays
+// one side from 1 copy and other side from other copy at higher zoom level if
+// screen crosses internation dateline
+function calculateBoundingBoxLngLimit(bounds: LngLatBounds) {
+  const boundsMinLng = bounds.getNorthWest().lng;
+  const boundsMaxLng = bounds.getSouthEast().lng;
+  // if bounds expands more than 360 then, consider complete globe is visible
+  if (boundsMaxLng - boundsMinLng >= MAX_LONGITUDE - MIN_LONGITUDE) {
```

Comment:
I think we should use absolute here to calculate the diff.

Reply:
Yeah. SouthWest will always be greater than NorthWest in Maplibre since we are not wrapping.